mhs/outbound/outbound/request/synchronous/tests/test_handler.py

```diff
@@ -215,10 +230,83 @@ def test_sync_async_workflow_not_invoked(self):
         self.workflow.handle_outbound_message.assert_called_once()
         self.assertEqual(expected_response.encode(), response.body)
+
+    def test_error_when_no_sync_async_header_present(self):
+        response = self.fetch("/", method="POST", headers={"Interaction-Id": INTERACTION_NAME},
+                              body=REQUEST_BODY)
+
+        self.assertEqual(response.code, 400)
+        self.assertIn("Sync-Async header missing", response.body.decode())
+
+    def test_error_when_sync_async_header_not_interactions(self):
+        self.config_manager.get_interaction_details.return_value = {'workflow': WORKFLOW_NAME}
+
+        response = self.fetch("/", method="POST", headers={"Interaction-Id": INTERACTION_NAME, 'sync-async': 'true'},
+                              body=REQUEST_BODY)
+
+        self.assertEqual(response.code, 500)
+        self.assertIn(" Failed to parse sync-async flag from interactions file for this interaction,"
+                      " contact your administrator",
+                      response.body.decode())
+
+    def test_correct_workflow_called(self):
+        self.setup_workflows()
+        self.config_manager.get_interaction_details.return_value = {'sync_async': 'false',
+                                                                    'workflow': WORKFLOW_NAME}
+        self.fetch("/", method="POST", headers={"Interaction-Id": INTERACTION_NAME, 'sync-async': 'false'},
+                   body=REQUEST_BODY)
+        self.workflow.handle_outbound_message.assert_called_once()
+
+    def test_sync_async_when_interactions_and_header_are_true(self):
+        self.setup_workflows()
+
+        self.config_manager.get_interaction_details.return_value = {'sync_async': True, 'workflow': WORKFLOW_NAME}
+        self.fetch("/", method="POST", headers={"Interaction-Id": INTERACTION_NAME, 'sync-async': 'true'},
+                   body=REQUEST_BODY)
+        self.sync_async_workflow.handle_sync_async_outbound_message.assert_called_once()
+
+    def test_error_when_interactions_is_false_and_header_is_true(self):
+        self.setup_workflows()
+        self.config_manager.get_interaction_details.return_value = {'sync_async': False, 'workflow': WORKFLOW_NAME}
+        response = self.fetch("/", method="POST", headers={"Interaction-Id": INTERACTION_NAME, 'sync-async': 'true'},
+                              body=REQUEST_BODY)
+
+        self.assertEqual(response.code, 400)
+        self.assertIn("Message header requested sync-async wrap for a message patternthat does not support sync-async",
+                      response.body.decode())
+        self.sync_async_workflow.handle_sync_async_outbound_message.assert_not_called()
+        self.workflow.handle_outbound_message.assert_not_called()
+
+    def test_not_sync_async_when_interactions_is_true_and_header_is_false(self):
+        self.setup_workflows()
+        self.config_manager.get_interaction_details.return_value = {'sync_async': 'true', 'workflow': WORKFLOW_NAME}
```

Comment:
In some places, you're returning sync_async as `'true'`, in other places as `True`. Should be consistent. In the actual file you're using a boolean, so I think in these tests a boolean should also be being used, if that's what it's meant to be.